### ISSUE TITLE
Priority in fluent properties

### ISSUE THREAD

**Issue by yloiseau** [eclipse/golo-lang#471]
We introduced fluent properties, that allows to call `setValue(v)` as just `value(v)` and `getValue()` or `isValue()` as `value()`. 
However, if a `value()` method exists, the property is called instead.

For instance, given:
```java
public class A {

  public String getValue() {
    return "getter";
  }

  public Integer value() {
    return 42;
  }
}
```
and
```golo
module Test

function main = |args| {
  println(A(): value())
}
```

`javac A.java && golo golo --files test.golo` will print `getter`, while `42` is expected

**Comment by danielpetisme** [comment]
Not sure it's a bug ....
The current [method resolution](findTarget) looks in order for:
* first a property. Given a property `name`, the [property finder](https://github.com/eclipse/golo-lang/blob/092cbdcce4d580bf1c096972c611ff23290595f1/src/main/java/org/eclipse/golo/runtime/PropertyMethodFinder.java#L51) looks for (in order) :
  * `getName`
  * `isName`
  * `setName`
* then for a method. Given a property `name`, it looks for a method named `name` + number of arguments required.
* finally look for a method in the class augmentation

So basically it's the normal behavior so now 2 options, either we say it's not a bug and we document the Method finding algo. or we modify the algo. 

For me current behavior sounds wise because the classic Java methods (the explicit ones) are used in the first place.

**Comment by yloiseau** [comment]
I think it qualifies as a bug, since in my example, there is no way to call the `value()` method.

**Comment by danielpetisme** [comment]
What would be the correct order? Method, property then augmentation ?

**Comment by yloiseau** [comment]
I think so. IMO we should involve the "magic" only when "normal" method fails. Don't know if others agree @Artpej @jponge

**Comment by yloiseau** [comment]
I'd say it can be considered bad design to have both a "bean" interface *and* a fluent one (as in my example) but it could be useful (e.g. for framework compatibility). Anyway, we should not make such objects unusable.
